chain-storage/src/store.rs

```diff
@@ -198,21 +225,29 @@ pub trait BlockStore<B>: std::marker::Sized where B: Block {
                     store: &self,
                     to_depth: to_info.depth,
                     cur_depth: to_info.depth - distance,
-                    pending_infos: vec![to_info]
+                    pending_infos: vec![to_info],
                 })
             }
         }
     }
 }
 
-pub struct BlockIterator<'store, B, S> where B: Block, S: BlockStore<B> + 'store {
+pub struct BlockIterator<'store, B, S>
+where
+    B: Block,
+    S: BlockStore<B> + 'store,
```

Comment:
Why must `BlockStore` have the block type as a parameter and not as an associated type? I think this percolates into excessive generic type parameters in other places, such as here.

Reply:
Thanks, changed it into an associated type.